docs/guides/threading.md

```diff
@@ -313,6 +313,17 @@ force quit your program.
 
 So whenever possible, sprinkle your long-running functions with `yield`.
 
+If you want your thread to quit when the napari viewer closes, you can connect
```

Comment:
I probably should have asked you in the previous issue _why_ you were trying to do this... because napari will automatically clean up threads started with `@thread_worker` at close, so I'm not sure that I agree that this is recommended.

The exception _might_ be in the case of multiple viewers in a single session, is that what you're doing?  Curious what made you look for a way to do this in the first place.  did you see something funny going on?  or were you trying to be a good citizen with the cleanup? :)

Reply:
relevant code is here: https://github.com/napari/napari/blob/87a61fb420c7b9f479433f7e5e4bc2378817d17e/napari/_qt/qthreading.py#L409-L487

and `wait_for_workers_to_quit` is automatically hooked up [here](https://github.com/napari/napari/blob/74dbbf23fb1deea5690810213cbe353e728801f5/napari/_qt/qt_event_loop.py#L175)

Reply:
My use case can be found [here](https://github.com/haesleinhuepf/napari-webcam/blob/master/napari_webcam/_dock_widget.py#L73) it was pretty tricky to make this work without an endless loop and crashes. I had to add the `while ...viewer...` thingy to make it work. Maybe, there is an alternative....

Reply:
and what happens if you simple make it `while True:`

Reply:
the main point here is that `# loop until napari closes` _is_ the default behavior... no special while loop or special shutdown should be necessary.  If that's not working for you, I'd like to know why

Reply:
In my case ([see branch](https://github.com/haesleinhuepf/napari-webcam/commit/b50410883c55fbbfcc7721121aa29067d581e37c)), the thread continues to run after closing napari. I have to use the Task Manager to kill the process...

Reply:
... I think I can make it work on my side. Maybe this documentation change is not necessary...

Reply:
Here's an example of a streaming image viewer, I just tested this on windows and it quits gracefully, even if the streaming is happening while you quit:

```python
from qtpy.QtWidgets import QPushButton, QVBoxLayout, QWidget
import napari
import time
from napari._qt.qthreading import thread_worker

import numpy as np


class FakeCamera(QWidget):
    def __init__(self, viewer: 'napari.viewer.Viewer'):
        super().__init__()
        self._viewer = viewer
        self.worker = None
        self.setLayout(QVBoxLayout())
        self.button = QPushButton("start streaming")
        self.button.clicked.connect(self.toggle_live)
        self.layout().addWidget(self.button)

    def _update_viewer(self, image):
        name = 'stream'
        try:
            layer = self._viewer.layers[name]
            layer.data = image
        except KeyError:
            self._viewer.add_image(image, name=name)

    def toggle_live(self):
        if self.worker is not None:
            self.worker.quit()
            self.worker = None
            self.button.setText('start streaming')
            return

        @thread_worker(connect={'yielded': self._update_viewer})
        def continuous_acquire():
            while True:
                yield np.random.rand(256, 256)
                time.sleep(0.05)
        
        self.button.setText('stop streaming')
        self.worker = continuous_acquire()
        

# create a viewer window
viewer = napari.Viewer()

viewer.window.add_dock_widget(FakeCamera(viewer))
# Start napari
napari.run()
```

so I'd like to first figure out what's going wrong in your branch there before adding this documentation.  Cause if there is some continuous behavior here, this is a bug (rather than something that needs documentation fixing)